lambdas/preview/index.py

```diff
@@ -141,35 +145,102 @@ def extract_vcf(file_):
     """
     Pull summary info from VCF: meta-information, header line, and data lines
     (in that order, up to MAX_LINES)
-    """
 
+    Args:
+        file_ - file-like
+    """
     meta = []
     header = []
     data = []
-
+    size = 0
     with open(file_.name, 'rb') as vcf:
         for index, line in enumerate(vcf, start=1):
-            # don't escape quotes
-            line = line.strip().decode('unicode_escape')
-            formatted = f'<p>{line}</p>'
+            line = _truncate(line.rstrip().decode(), size)
```

Comment:
add an explicit decoding scheme as an argument to decode. 

I'm also a little surprised that rstrip works on a bytes object and that it shouldn't be `line.decode().rstrip()` instead

Reply:
also, this line is a little confusing because it switched the type of line via a function

Reply:
Can we just open the file as `'r'` instead of `'rb'`? Or is byte counting that important?

Reply:
byte strings better for size counting. doc-string notes that, in some rare cases (e.g. a file on a really long line) we may break a unicode character mid-stream. but usually this is not an issue (we don't usually break lines up).

Reply:
actually... i think we can

Reply:
i looked into this and it 1) breaks the parquet unit test (where we don't even touch encoding); 2) breaks all but one other test and, at first glance, did not find a simple fix.